### ISSUE TITLE
Suggestions on how to `reverse` in my Django settings.yaml

### ISSUE THREAD

**Issue by wgordon17** [dynaconf/dynaconf#858]
I currently define settings like `LOGIN_URL` as `reverse_lazy("account_login")`. Is there a way to replicate this type of configuration into something I can use in my Dynaconf settings.yaml?

If possible, I'd prefer to keep all settings in my Dynaconf settings.yaml and keep my Django settings.py file as empty as possible.

**Comment by rochacbruno** [comment]
What you can do right now is the following:

```python
# on the top import reverse_lazy
from django.urls import reverse_lazy


Then after the `import dynaconf`  - before `settings = ...` add

```pythin
dynaconf.utils.parse_conf["@reverse_lazy"] = reverse_lazy
```


Now you should be able to do

On yaml
```yaml
default:
    login_url: "@reverse_lazy account_login"
```

On env vars
```bash
export DJANGO_LOGIN_URL="@reverse_lazy account_login"
```

**NOTE** I have not tested! but if it works we can then add it by default to the context in future versions.

**Comment by wgordon17** [comment]
This is _excellent_, thank you! 🎉 

So, based on the format in https://www.dynaconf.com/envvars/#adding-a-custom-casting-token and from your suggestion, I came up with

```python
# HERE STARTS DYNACONF EXTENSION LOAD (Keep at the very bottom of settings.py)
# Read more at https://www.dynaconf.com/django/
import dynaconf  # noqa
from dynaconf.utils import parse_conf  # noqa

parse_conf.converters["@reverse_lazy"] = (
    lambda value: value.set_casting(reverse_lazy)
    if isinstance(value, parse_conf.Lazy)
    else reverse_lazy(value)
)
settings = dynaconf.DjangoDynaconf(__name__)  # noqa
# HERE ENDS DYNACONF EXTENSION LOAD (No more code below this line)
```

And then I was able to do exactly what you suggestion in my `settings.yaml`

```yaml
default:
    LOGIN_URL: "@reverse_lazy account_login"
```

Perfect! 👌

**Comment by wgordon17** [comment]
I have _no_ idea what was going on when I first validated this, but I have _yet_ to be able to reproduce getting this working. After _waayyy_ too much time debugging this, I discovered that https://github.com/dynaconf/dynaconf/blob/master/dynaconf/base.py#L894 is the offending line breaking this functionality

```python
if existing is not None and existing != value:
```

Specifically, this little piece here, `existing != value`. When comparing the `__proxy__` value, returned by `reverse_lazy`, it was triggering an evaluation in that moment, and the evaluation was failing because the settings weren't fully configured yet, and specifically the `ROOT_URLCONF` was failing to be found 🤦‍♂️ 

Ok, with all of that said, it turns out it's a relatively simple fix that _is_ working now 👍 

```python
parse_conf.converters["@reverse_lazy"] = lambda value: parse_conf.Lazy(value, casting=reverse_lazy)
```

I'd be _very_ curious to know if this is an anti-pattern, or something liable to break? By _explicitly_ creating this as a `Lazy` class, I'm able to defer evaluation in the above `base.py` file, while still being able to evaluate to the desired path lazily.
